tools/Raven.Migrator/CosmosDB/CosmosDBMigrator.cs

```diff
@@ -0,0 +1,171 @@
+            string ravenCollectionName,
+            Reference<bool> isFirstDocument,
+            StreamWriter streamWriter)
+        {
+            var collection = UriFactory.CreateDocumentCollectionUri(database.Id, cosmosCollectionName);
+            var options = new FeedOptions
+            {
+                MaxItemCount = 128
+            };
+
+            var query = client.CreateDocumentQuery<ExpandoObject>(collection, options).AsDocumentQuery();
+            var attachmentNumber = 0;
+
+            while (query.HasMoreResults)
+            {
+                var result = await query.ExecuteNextAsync<ExpandoObject>();
+                foreach (var document in result)
+                {
+                    var dictionary = (IDictionary<string, object>)document;
+                    var documentId = dictionary[CosmosDocumentId].ToString();
+
+                    const string selfLinkKey = "_self";
+                    var selfLink = dictionary[selfLinkKey].ToString();
+
+                    var attachments = new List<Dictionary<string, object>>();
+                    foreach (var attachment in client.CreateAttachmentQuery(selfLink))
+                    {
+                        var attachmentUri = UriFactory.CreateAttachmentUri(database.Id, cosmosCollectionName, documentId, attachment.Id);
+                        var attachmentResponse = await client.ReadAttachmentAsync(attachmentUri);
+                        var resourceMediaLink = attachmentResponse.Resource.MediaLink;
+                        var mediaLinkResponse = await client.ReadMediaAsync(resourceMediaLink);
+
+                        attachmentNumber++;
+
+                        var attachmentInfo = await MigrationHelpers.WriteAttachment(
+                            mediaLinkResponse.Media, mediaLinkResponse.Media.Length,
+                            attachmentResponse.Resource.Id, ravenCollectionName,
+                            mediaLinkResponse.ContentType,attachmentNumber, isFirstDocument, streamWriter);
+
+                        attachments.Add(attachmentInfo);
+                    }
+
+                    await MigrationHelpers.WriteDocument(
+                        document,
+                        documentId,
+                        ravenCollectionName,
+                        PropertiesToRemove,
+                        isFirstDocument,
+                        streamWriter,
+                        attachments);
+                }
+            }
+        }
+
+        private static Dictionary<string, string> GetCollectionsToMigrate(DocumentClient client, Database database)
+        {
+            var collectionsToMigrate = new Dictionary<string, string>();
+
+            var azureCollections = client.CreateDocumentCollectionQuery(database.SelfLink).ToList();
+            foreach (var azureCollection in azureCollections)
+            {
+                collectionsToMigrate.Add(azureCollection.Id, azureCollection.Id);
```

Comment:
you can simply this using `client.CreateDocumentCollectionQuery(database.SelfLink).ToDictionary()`